common/util/util.go

```diff
@@ -172,3 +173,14 @@ func RepeatSlice[T any](xs []T, n int) []T {
 func Ptr[T any](v T) *T {
 	return &v
 }
+
+func IsGenericNil[T any](val T) bool {
```

Comment:
I kinda don't like this version.. if you're checking nil like that I think it usually indicates you're doing something wrong with reflection. and in the rare case you want to do that, you're probably already know your Kind or are dealing with different kinds differently

Reply:
Well we expect our persistence managers to always have the interface kind, but I don't know that we can enforce that, so I thought it was better to include all the possibilities